maskrcnn_benchmark/data/datasets/evaluation/voc/voc_eval.py

```diff
@@ -0,0 +1,220 @@
+# A modification version from chainercv repository.
+# (See https://github.com/chainer/chainercv/blob/master/chainercv/evaluations/eval_detection_voc.py)
+from __future__ import division
+
+import itertools
+import os
+import torch
+from collections import defaultdict
+import numpy as np
+from maskrcnn_benchmark.structures.bounding_box import BoxList
+from maskrcnn_benchmark.structures.boxlist_ops import boxlist_iou
+
+
+def do_voc_evaluation(dataset, predictions, output_folder, logger):
+    pred_boxlists = []
+    gt_boxlists = []
+    for image_id, prediction in enumerate(predictions):
+        img_info = dataset.get_img_info(image_id)
+        original_id = dataset.id_to_img_map[image_id]
+        if len(prediction) == 0:
+            continue
+        image_width = img_info['width']
+        image_height = img_info["height"]
+        prediction = prediction.resize((image_width, image_height))
+        pred_boxlists.append(prediction)
+
+        ann = dataset.anns[original_id]
+        gt_boxes = ann['boxes']
+        gt_labels = ann['labels']
+        gt_boxlist = BoxList(gt_boxes, image_size=(image_width, image_height))
+        gt_boxlist.add_field('labels', torch.as_tensor(gt_labels))
+        gt_boxlists.append(gt_boxlist)
+    result = eval_detection_voc(pred_boxlists=pred_boxlists, gt_boxlists=gt_boxlists, use_07_metric=True)
+    result_str = 'mAP: {:.4f}\n'.format(result['map'])
+    for i, ap in enumerate(result['ap']):
+        if i == 0:  # skip background
+            continue
+        result_str += '{:<16}: {:.4f}\n'.format(dataset.map_class_id_to_class_name(i), ap)
+    logger.info(result_str)
+    if output_folder:
+        with open(os.path.join(output_folder, 'result.txt'), 'w') as fid:
+            fid.write(result_str)
+    return result
+
+
+def eval_detection_voc(pred_boxlists,
+                       gt_boxlists,
+                       iou_thresh=0.5,
+                       use_07_metric=False):
+    """Evaluate on voc dataset.
+    Args:
+        pred_boxlists(list[BoxList]): pred boxlist, has labels and scores fields.
+        gt_boxlists(list[BoxList]): ground truth boxlist, has labels field.
+        iou_thresh: iou thresh
+        use_07_metric: boolean
+    Returns:
+        dict represents the results
+    """
+    assert len(gt_boxlists) == len(pred_boxlists), 'Length of gt and pred lists need to be same.'
+    prec, rec = calc_detection_voc_prec_rec(pred_boxlists=pred_boxlists,
+                                            gt_boxlists=gt_boxlists,
+                                            iou_thresh=iou_thresh)
```

Comment:
You still need to pass `gt_difficults` in here if you want it to be evaluated properly